File: kushki-android/src/main/java/com/kushkipagos/android/Transaction.java
package com.kushkipagos.android;

import org.json.JSONException;
import org.json.JSONObject;

public class Transaction {

    private final String code;
    private final String message;
    private final String token;
    private final boolean successful;

    Transaction(String responseBody) {
        try {
            JSONObject jsonResponse = new JSONObject(responseBody);
            code = jsonResponse.getString("response_code");
            message = jsonResponse.getString("response_text");
            token = jsonResponse.getString("transaction_token");
            successful = "000".equals(code);
        } catch (JSONException jsonException) {
            throw new IllegalArgumentException(jsonException);
        }
    }

    public String getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getToken() {
        return token;
    }

    public boolean isSuccessful() {
        return successful;
    }
}

File: kushki-android/src/main/java/com/kushkipagos/android/Transaction.kt
package com.kushkipagos.android

import org.json.JSONException
import org.json.JSONObject

class Transaction(responseBody: String) {

    var code: String
    var message: String
    var token: String = ""
    var settlement: Double = 0.0
    var secureId: String
    var secureService: String
    var isSuccessful: Boolean = false
    val jsonResponse: JSONObject = JSONObject(responseBody)

    init {
        settlement = try {
            jsonResponse.getDouble("settlement")
        } catch (jsonException: JSONException) {
            0.0
        }
        secureId = try {
            jsonResponse.getString("secureId")
        } catch (jsonException: JSONException) {
            ""
        }
        secureService = try {
            jsonResponse.getString("secureService")
        } catch (jsonException: JSONException) {
            ""
        }
        try {
            code = "000"
            message = ""
            token = jsonResponse.getString("token")

            if (token != "") isSuccessful = "000" == code
            else {
                code = jsonResponse.getString("code")
                message = jsonResponse.getString("message")
            }
        } catch (jsonException: JSONException) {
            code = jsonResponse.getString("code")
            message = jsonResponse.getString("message")
        }

    }
}
